adam_core/observations/exposures.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from typing import Iterator, Literal
+
+import astropy.time
+import pyarrow as pa
+import pyarrow.compute as pc
+import quivr as qv
+from quivr.validators import and_, ge, le
+
+from ..coordinates import cartesian, origin, times
+from ..observers import state
+
+
+class Exposures(qv.Table):
+    """
+    Exposures is a table of data about exposures that provide point source observations.
+    """
+
+    id = qv.StringColumn()
+    start_time = times.Times.as_column()
+    duration = qv.Float64Column(validator=and_(ge(0), le(3600)))
+    filter = qv.DictionaryColumn(index_type=pa.uint32(), value_type=pa.string())
+
+    observatory_code = qv.StringColumn()
+
+    def group_by_observatory_code(self) -> Iterator[tuple[str, Exposures]]:
+        """
+        Groups the exposures by observatory code.
+        """
+        unique_codes = self.observatory_code.unique()
+        for code in unique_codes:
+            mask = pc.equal(self.observatory_code, code)
+            yield code, self.apply_mask(mask)
+
+    def observer_states(
+        self,
+        frame: Literal["ecliptic", "equatorial"] = "ecliptic",
+        origin: origin.OriginCodes = origin.OriginCodes.SUN,
+    ) -> cartesian.CartesianCoordinates:
```

Comment:
I was *really* feeling the power of adam_core when I wrote this method. Nice job!